docs/psa_switch-dev.md

```diff
@@ -0,0 +1,82 @@
+# Implementing Externs in PSA Switch
+Externs in PSA Switch are different than those in Simple Switch. 
+Simple Switch uses externs baked into the BMV2 model where as PSA Switch
+uses externs specific to its own architecture. To understand what this
+translates to in code, look at the counter used in 
+[Simple Switch](https://github.com/p4lang/behavioral-model/tree/master/src/bm_sim)
+and the one in 
+[PSA Switch](https://github.com/p4lang/behavioral-model/tree/master/targets/psa_switch/externs).
+The Simple Switch counter lives inside of `src/bm_sim` where as the 
+counter for PSA Switch lives inside of the target specific directory.
+This doc is meant to provide guidance for developers on how to develop
+new externs in PSA Switch.
+
+## Background Knowledge
+For those who are familiar with the internals of BMV2 feel free
+to skip this section. For those who are new/less familiar with how BMV2
+operates, consider reading `P4Objects.cpp` as a starting point
+as this is where externs will be created. From here you can also see
+what the expected JSON for externs should be. Next, read the changes
+in the following [commit](https://github.com/p4lang/behavioral-model/pull/767).
+It details all the changes made to support the counter extern in PSA.
+It will also be useful to understand on a high level how P4 Runtime
+interacts with PSA Switch. Briefly skimming `runtime_CLI.py` will do.
+
+## High-Level Process
+The following process is meant to serve as a general guide on how
+to develop new externs in PSA Switch.
+
+1. Provide support for emitting the extern in p4c. Currently, only
```

Comment:
Something that is worth mentioning because it explains the design we came up with: IIRC p4c is still generating "built-in" objects as part of the JSON (they are unused by PSASwitch) in order to make sure that runtime tools which consume the bmv2 JSON (such as the runtime_CLI) still work. This is temporary until such tools can be updated. Please correct me if I'm wrong.